lua/nvim-tree/utils.lua

```diff
@@ -68,13 +79,18 @@ function M.path_relative(path, relative_to)
     return path
   end
 
-  local _, r = path:find(M.path_add_trailing(relative_to), 1, true)
-  local p = path
+  local norm_path = path
+  if has_parentheses_and_brackets(path) then
+    norm_path = path:gsub("/", "\\")
+  end
+
+  local _, r = norm_path:find(M.path_add_trailing(relative_to), 1, true)
+  local p = norm_path
```

Comment:
This path is executed for all OS, which may cause issues for such paths. Please execute only when `is_windows`

This injected error is definitely thrown on linux, showing that the code path is executed:
```lua
--- Check if there are parentheses before brackets, it causes problems for windows.
--- Refer to issue #2862 and #2961 for more details.
local function has_parentheses_and_brackets(path)
  error("has_parentheses_and_brackets")
```
```
E5108: Error executing lua: ...d/site/pack/packer/start/ljie-PI/lua/nvim-tree/utils.lua:65: has_parentheses_and_brackets called
stack traceback:
        [C]: in function 'error'
        ...d/site/pack/packer/start/ljie-PI/lua/nvim-tree/utils.lua:65: in function 'has_parentheses_and_brackets'
        ...d/site/pack/packer/start/ljie-PI/lua/nvim-tree/utils.lua:84: in function 'path_relative'
        .../packer/start/ljie-PI/lua/nvim-tree/explorer/filters.lua:161: in function 'custom'
        .../packer/start/ljie-PI/lua/nvim-tree/explorer/filters.lua:247: in function 'should_filter_as_reason'
```

Reply:
Added flag on line 84, so that norm_path will keep unchanged for non-windows OS.